pkg/model/iam/iam_builder.go

```diff
@@ -962,48 +1002,15 @@ func AddAWSEBSCSIDriverPermissions(p *Policy, partition string, appendSnapshotPe
 		"ec2:DeleteVolume",            // aws.go
 		"ec2:DetachVolume",            // aws.go
 	)
-	p.clusterTaggedCreateAction.Insert(
-		"ec2:CreateVolume", // aws.go
-	)
 
-	p.Statement = append(p.Statement,
-		&Statement{
-			Effect: StatementEffectAllow,
-			Action: stringorslice.String(
-				"ec2:CreateTags", // aws.go, tag.go
-			),
-			Resource: stringorslice.Slice(
-				[]string{
-					fmt.Sprintf("arn:%v:ec2:*:*:volume/*", partition),
-					fmt.Sprintf("arn:%v:ec2:*:*:snapshot/*", partition),
-				},
-			),
-			Condition: Condition{
-				"StringEquals": map[string]interface{}{
-					"ec2:CreateAction": []string{
-						"CreateVolume",
-						"CreateSnapshot",
-					},
-				},
-			},
+	p.AddEC2CreateAction(
+		[]string{
+			"CreateVolume",
+			"CreateSnapshot",
 		},
-
-		&Statement{
-			Effect: StatementEffectAllow,
-			Action: stringorslice.String(
-				"ec2:DeleteTags", // aws.go, tag.go
-			),
-			Resource: stringorslice.Slice(
-				[]string{
-					fmt.Sprintf("arn:%v:ec2:*:*:volume/*", partition),
-					fmt.Sprintf("arn:%v:ec2:*:*:snapshot/*", partition),
-				},
-			),
-			Condition: Condition{
-				"StringEquals": map[string]string{
-					"aws:ResourceTag/KubernetesCluster": p.clusterName,
-				},
-			},
+		[]string{
+			fmt.Sprintf("arn:%v:ec2:*:*:volume/*", partition),
+			fmt.Sprintf("arn:%v:ec2:*:*:snapshot/*", partition),
```

Comment:
Perhaps some of this partition logic could be pushed down?

Reply:
partition should probably be on the policy object itself. but that seems like something for a follow-up.